gpt_slide_expander.py

```diff
@@ -0,0 +1,177 @@
+
+    @staticmethod
+    def configure_api_key():
+        """Configure OpenAI API key."""
+        openai.api_key = os.getenv("OPENAI_API_KEY")
+
+    @staticmethod
+    def generate_presentation_main_topic(presentation_text: str, max_retry: int) -> str:
+        """
+        Generates the main topic of a presentation based on the presentation's text.
+        Args:
+            presentation_text (str): The text of the presentation.
+            max_retry (int): The maximum number of retries for generating the main topic.
+        Returns:
+            str: The generated main topic of the presentation.
+        Raises:
+            RuntimeError: If the query fails after reaching the maximum number of retries.
+            ValueError: If the presentation text is empty or invalid.
+        """
+        if presentation_text:
+            retry_count = 0
+            wait_time = 4  # Initial wait time in seconds
+
+            while retry_count < max_retry:
+                try:
+                    gpt_response = openai.ChatCompletion.create(
+                        model="gpt-3.5-turbo",
+                        messages=[{"role": "system",
+                                   "content": "You are a chatbot that extracts the main topic of a presentation"
+                                              " based on the presentation's text."},
+                                  {"role": "user",
+                                   "content": f"Provide the main topic of this presentation based on the presentation's"
+                                              f" following text: {presentation_text} , no longer than one word."}]
+
+                    )
+
+                    return gpt_response['choices'][0]['message']['content']
+
+                except openai.error as e:
+                    pass  # error will be documented in log files in future
+
+                # Increase wait time and retry
+                retry_count += 1
+                time.sleep(wait_time)
+                wait_time += 1
+
+            raise RuntimeError(f"Query failed after {max_retry} attempts.")
+
+        else:
+            raise ValueError("Invalid data. presentation's text cannot be empty.")
+
+    async def generate_explanation_for_presentation(self, parsed_presentation: Dict[int, str], presentation_topic: str,
+                                                    max_retry: int) -> None:
+        """
+        Generates explanations for each slide in a presentation.
+        Args:
+            parsed_presentation (Dict[int, str]): A dictionary mapping slide index to slide content.
+            presentation_topic (str): main topic of presentation.
+            max_retry (int): The maximum number of retries for generating an explanation for each slide.
+        Raises:
+            RuntimeError: If the main topic retrieval from the API fails.
+        Returns:
+            None
+        """
+        # Generate tasks for each slide
```

Comment:
The comments arent needed, python syntax and valid variable naming explains it well enough